swap_meet/vendor.py

```diff
@@ -1,2 +1,114 @@
+            return False
+        else:
+            self.inventory.remove(item)
+            return item   
+         
+    def get_by_id(self, id): 
+        # item = Item(id)   
+        for item in self.inventory:
+            if id == item.id:
+                return item
+        
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if (my_item not in self.inventory 
+            or their_item not in other_vendor.inventory):
+            return False
+        else:
+            self.remove(my_item)
+            other_vendor.add(my_item)
+            self.add(their_item)
+            other_vendor.remove(their_item)
+            return True
+        
+    def swap_first_item(self, other_vendor):
+        if self.inventory == [] or other_vendor.inventory == []:
+            return False
+        other_vendor.inventory[0], self.inventory[0] = self.inventory[0], other_vendor.inventory[0] 
+        return True 
+    
+    def get_by_category(self, category):
+        if self.inventory == []:
+            return []
+        list_by_category =[]
+        for item in self.inventory:
+            if item.get_category() == category:
+                list_by_category.append(item) 
+        return list_by_category        
+
+        
+
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+        else:
+            return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        #Checks if items are located in the correct starting vendor
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.remove(my_item)
+            other_vendor.remove(their_item)
+            self.add(their_item)
+            other_vendor.add(my_item)
+            return True
+        else:
+            return False
+    
+    def swap_first_item(self, other_vendor):
+        if self.inventory == [] or other_vendor.inventory == []:
+            return False
+        else:
+            my_item = self.inventory[0]
+            their_item = other_vendor.inventory[0]
+
+            self.remove(my_item)
+            other_vendor.remove(their_item)
+
+            self.add(their_item)
+            other_vendor.add(my_item)
+
+            return True
+    
+    def get_by_category(self, category):
+        if self.inventory == []:
+            return []
+        category_list = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                category_list.append(item)
+        return category_list
+    
+    def get_best_by_category(self, category):
+        category_list = self.get_by_category(category)
+        if category_list == []:
+            return None
+        best_score = 0
+        for item in category_list:
+            if item.condition > best_score:
+                best_score = item.condition
+                best_item = item
```

Comment:
👍 This might be the last time you need to manually write a "max" loop. After this project, keep an eye out for where you can make use of `max` and similar functions!